### PR TITLE
Update neutrino binary with latest bitcoin core code

### LINKED ISSUE #1629: Update neutrino binary with latest bitcoin core code
Our neutrino binary is out of date, we should update to the latest functionality and test it

https://github.com/bitcoin/bitcoin/pull/18876

### PR DESCRIPTION
**Pull request by rorp**
Fixes https://github.com/bitcoin-s/bitcoin-s/issues/1629

### CODE REVIEW

File: project/CommonSettings.scala [hunk 1]
```diff
@@ -77,7 +77,8 @@ object CommonSettings {
     )
   }
 
-  private val scala2_13CompilerOpts = Seq("-Xlint:unused", "-Xfatal-warnings")
+  private val scala2_13CompilerOpts =
+    Seq("-Xlint:unused") //, "-Xfatal-warnings")
```
Comment (benthecarman) [inline project/CommonSettings.scala]: need to revert these

**Comment by benthecarman** [comment]
I think this link needs to be changed as well?

https://github.com/bitcoin-s/bitcoin-s/blob/master/bitcoind-rpc/bitcoind-rpc.sbt#L53

**Comment by Christewart** [comment]
It looks like it's failing from RPC level changes? Any insight @benthecarman ? https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/733058582#L486

**Comment by benthecarman** [comment]
> It looks like it's failing from RPC level changes? Any insight @benthecarman ? [travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/733058582#L486](https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/733058582#L486)

Pushed a fixed, seemed our `GetMempoolResult`s were wrong for v18

**Comment by benthecarman** [comment]
Looks like something is wrong with the mac binaries?

https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/733133600#L2691

**Comment by Christewart** [comment]
Now seems to have problems with a bloom filter rpc? 

https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/733133601#L597

Also there seems to be a bunch of static linking errors with the v18 bitcoind binary? Shouldn't this be just the stock v18 release from bitcoin core now? 

https://travis-ci.org/github/bitcoin-s/bitcoin-s/jobs/733133600#L2717

**Pull request closed without merging**